src/appleseed/renderer/kernel/lighting/lighttypes.cpp

```diff
@@ -161,6 +194,20 @@ void EmittingShape::sample_uniform(
         // Set the world space geometric normal.
         light_sample.m_geometric_normal = m_geom.m_triangle.m_geometric_normal;
     }
+    else if (shape_type == RectangleShape)
+    {
+        // Set the barycentric coordinates.
+        light_sample.m_bary = s;

```

Comment:
Maybe not for this PR, but it would make sense to rename m_bary now that they are not always barycentric coordinates.
Maybe m_param?